pkg/asset/installconfig/gcp/validation.go

```diff
@@ -133,3 +134,25 @@ func validateMachineNetworksContainIP(fldPath *field.Path, networks []types.Mach
 	}
 	return field.ErrorList{field.Invalid(fldPath, subnetName, fmt.Sprintf("subnet CIDR range start %s is outside of the specified machine networks", ip))}
 }
+
+func CheckForServicesToBeEnabled(projectServices []string) error {
+	services := sets.NewString("compute.googleapis.com", "cloudapis.googleapis.com",
+		"cloudresourcemanager.googleapis.com", "dns.googleapis.com",
+		"iam.googleapis.com", "iamcredentials.googleapis.com",
+		"servicemanagement.googleapis.com","serviceusage.googleapis.com",
+		"storage-api.googleapis.com", "storage-component.googleapis.com")
+
+	if len(projectServices) == 0 {
+		return errors.New("there are no enabled services present")
+	}
+	var count int = 0
+	for _, service := range projectServices {
+		if services.Has(service) {
+			count++
```

Comment:
this approach is reasonable and works. 

To me, a more intuitive approach would be to make a set of the project's enabled services and iterate through the required services, checking if each is enabled. 

This would have the added benefit that if a required service was not found, you could append the missing service to a slice then include the slice of all missing services in an error message to the user, so they could see which service needs to be enabled.

Reply:
Also in the same area for improvement https://github.com/openshift/installer/pull/3773#discussion_r442583156